modules/basegl/src/processors/imageprocessing/imagegradient.cpp

```diff
@@ -86,12 +80,23 @@ ImageGradient::ImageGradient()
     });
 
     addProperty(channel_);
-    addProperty(renormalization_);
 }
 
 void ImageGradient::preProcess(TextureUnitContainer&) {
+    const auto layer = inport_.getData()->getColorLayer();
+    const auto gradientSpacing{layer->getWorldSpaceGradientSpacing()};
+    shader_.setUniform("worldSpaceGradientSpacing", gradientSpacing);
+    shader_.setUniform("textureSpaceGradientSpacing",
+                       mat2{glm::scale(layer->getCoordinateTransformer().getWorldToTextureMatrix(),
+                                       vec3{gradientSpacing, 1.0f})});
+
     shader_.setUniform("channel", channel_.getSelectedValue());
-    shader_.setUniform("renormalization_", renormalization_.get() ? 1 : 0);
+
+    const double gradientEstimate = glm::compMax(glm::abs(layer->dataMap.dataRange)) /
```

Comment:
Should be value range?
```suggestion
    const double gradientEstimate = glm::compMax(glm::abs(layer->dataMap.valueRange)) /
```

Reply:
Should be the data range for the calculation.

Reply:
I don't see why, could you elaborate? The dataRange has no physical meaning.